pkg/osquery/tables/xrdb/xrdb.go

```diff
@@ -0,0 +1,183 @@
+// +build linux
+
+package xrdb
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+var xrdbPath = "/usr/bin/xrdb"
+
+const allowedUsernameCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-."
+const allowedDisplayCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789:"
+
+type execer func(ctx context.Context, display, username string, buf *bytes.Buffer) error
+
+type XRDBSettings struct {
+	client   *osquery.ExtensionManagerClient
+	logger   log.Logger
+	getBytes execer
+}
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("key"),
+		table.TextColumn("value"),
+		table.TextColumn("display"),
+		table.TextColumn("username"),
+	}
+
+	t := &XRDBSettings{
+		client:   client,
+		logger:   logger,
+		getBytes: execXRDB,
+	}
+
+	return table.NewPlugin("kolide_xrdb", columns, t.generate)
+}
+
+func (t *XRDBSettings) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	users := tablehelpers.GetConstraints(queryContext, "username", tablehelpers.WithAllowedCharacters(allowedUsernameCharacters))
+	if len(users) < 1 {
+		return results, errors.New("kolide_xrdb requires at least one username to be specified")
```

Comment:
I forget how this works. we might also need a `level.Info(t.logger).Log(...)` here

Reply:
I can certainly add one, but I checked other tables and we don't log alongside the error return. 
The `kolide_cryptsetup_status` table is a recent example that still bubbles up an error (in verbose mode) when a constraint isn't specified: 

```
osquery> select * from kolide_cryptsetup_status;
I0614 12:52:34.403594 39429 virtual_table.cpp:1020] Invalid response from the extension table. Error 1: error generating table: The kolide_cryptsetup_status table requires that you specify a constraint for name
Error: error generating table: The kolide_cryptsetup_status table requires that you specify a constraint for name
```

Reply:
Good enough then. No need.